pkg/test/logger/logger.go

```diff
@@ -0,0 +1,30 @@
+package logger
+
+import (
+	"github.com/sirupsen/logrus"
+	logrustest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+)
+
+// NewLoggerWithHook creates a new logger with debug loglevel and attaches a hook to it.
+func NewLoggerWithHook() (*logrus.Logger, *logrustest.Hook) {
+	logger := logrus.New()
+	logger.SetLevel(logrus.DebugLevel)
+	hook := logrustest.NewLocal(logger)
+	return logger, hook
+}
+
+func AssertHookContainsMessage(t assert.TestingT, hook *logrustest.Hook, message string) bool {
+	var messages []string
+
+	if message == "" {
+		return true
+	}
+
+	if hook != nil {
+		for _, log := range hook.AllEntries() {
+			messages = append(messages, log.Message)
```

Comment:
When I wrote this block, I was trying to keep it simple knowing this was out of band of the functional code. If we want to tackle this, I agree it would be good to write a more efficient Assertion function here instead of using the existing building blocks. IMHO: Low hanging fruit, low priority.

Reply:
No need, I provided a [solution](https://github.com/openshift/hive/compare/6829e6e84d43ba416e416acc180a25ea51d5d748..2086ed265af068a6569085543f2e1d43a74597ed). When the message is found as expected, will just iterate the list and return true when found. However, if the message is not found then it will make a copy of the messages for outputting with Fail(). This is extremely helpful when unit tests are failing on a string comparison so we don't have to guess what is actually being logged. That way we get the best of both worlds.

Reply:
Cool. We still don't need to make an extra copy here, though. We can just use `hook.AllEntries()` as the format arg in the failure message.

Reply:
That is a fair point. It's not as clean output, but the information is still there. Better yet, just use `hook.Entries` since `hook.AllEntries()` actually makes a copy as well. [Done](https://github.com/openshift/hive/compare/2086ed265af068a6569085543f2e1d43a74597ed..944f18b7de7945ac146a5f7c2cd68553ea30e5c6).

Reply:
Cool. `hook.Entries` [says](https://github.com/openshift/hive/blob/2086ed265af068a6569085543f2e1d43a74597ed/vendor/github.com/sirupsen/logrus/hooks/test/test.go#L14-L17) it shouldn't be used directly, for thread safety, but in this scenario I think it's fine.